### PR TITLE
Change GIC init - create gic without ITS, use custom addresses

### PR DESCRIPTION
**Pull request by NunoDasNeves**

### CODE REVIEW

**Comment by russell-islam** [comment]
Looks like a lot of fmt and Clippy errors in both of the PRs. I will create another PR to address those issues.

File: arch/src/aarch64/mod.rs [hunk 1]
```diff
@@ -200,6 +200,10 @@ pub fn initramfs_load_addr(
 static mut KERNEL_START: u64 = layout::KERNEL_START;
 static mut RAM_64BIT_START: u64 = layout::RAM_64BIT_START;
 static mut FDT_START: u64 = layout::FDT_START;
+static mut GIC_V3_DIST_START: u64 = layout::GIC_V3_DIST_START;
+static mut GIC_V3_REDISTS_ADDR: u64 = layout::GIC_V3_DIST_START - layout::GIC_V3_REDIST_SIZE;
+static mut GIC_V3_REDISTS_SIZE: u64 = layout::GIC_V3_REDIST_SIZE;
+static mut GIC_V3_ITS_ADDR: u64 = unsafe{GIC_V3_REDISTS_ADDR - layout::GIC_V3_ITS_SIZE};
```
Comment (liuw) [inline arch/src/aarch64/mod.rs:206]: Not directly relate to this patch, but potentially we can have a `layout` structure that get initantiated according to some configurations (device tree, command line).

**Review by liuw — APPROVED** [review]
It is not perfect but it gets us closer to our goal.

File: arch/src/aarch64/mod.rs [hunk 2]
```diff
@@ -238,6 +266,22 @@ fn get_fdt_addr() -> u64 {
     unsafe{FDT_START}
 }
 
+pub fn get_gic_dist_addr() -> u64 {
+    unsafe {GIC_V3_DIST_START}
```
Comment (russell-islam) [inline arch/src/aarch64/mod.rs:270]: { GIC_V3_DIST_START }

File: arch/src/aarch64/mod.rs [hunk 3]
```diff
@@ -238,6 +266,22 @@ fn get_fdt_addr() -> u64 {
...
+}
+
+pub fn get_gic_redists_addr() -> u64 {
+    unsafe {GIC_V3_REDISTS_ADDR}
```
Comment (russell-islam) [inline arch/src/aarch64/mod.rs:274]: Same as above.

File: arch/src/aarch64/mod.rs [hunk 4]
```diff
@@ -238,6 +266,22 @@ fn get_fdt_addr() -> u64 {
...
+}
+
+pub fn get_gic_redists_size() -> u64 {
+    unsafe {GIC_V3_REDISTS_SIZE}
```
Comment (russell-islam) [inline arch/src/aarch64/mod.rs:278]: Same as above.

File: arch/src/aarch64/mod.rs [hunk 5]
```diff
@@ -238,6 +266,22 @@ fn get_fdt_addr() -> u64 {
...
+}
+
+pub fn get_gic_its_addr() -> u64 {
+    unsafe {GIC_V3_ITS_ADDR}
```
Comment (russell-islam) [inline arch/src/aarch64/mod.rs:282]: Same as above.

**Review by russell-islam — APPROVED** [review]

**Pull request closed without merging**